packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,339 @@
+import * as React from 'react';
+import { IDraggableZoneProps, ICoordinates, IDragData } from './DraggableZone.types';
+import { getClassNames } from './DraggableZone.styles';
+import { BaseComponent } from '../../Utilities';
+
+export interface IDraggableZoneState {
+  isDragging: boolean;
+  position: ICoordinates;
+  lastPosition?: ICoordinates;
+}
+
+const eventMapping = {
+  touch: {
+    start: 'touchstart',
+    move: 'touchmove',
+    stop: 'touchend'
+  },
+  mouse: {
+    start: 'mousedown',
+    move: 'mousemove',
+    stop: 'mouseup'
+  }
+};
+
+// These are needed so that we can generalize the events
+// and so we have access to clientX and clientY in the touch events
+type MouseTouchEvent<T> = React.MouseEvent<T> & React.TouchEvent<T>;
+
+export class DraggableZone extends BaseComponent<IDraggableZoneProps, IDraggableZoneState> {
+  private _touchId?: number;
+  private _currentEventType = eventMapping.mouse;
+
+  constructor(props: IDraggableZoneProps) {
+    super(props);
+
+    this.state = {
+      isDragging: false,
+      position: this.props.position || { x: 0, y: 0 },
+      lastPosition: undefined
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IDraggableZoneProps) {
+    if (nextProps.position && (!this.props.position || nextProps.position !== this.props.position)) {
```

Comment:
No tests for this component.

Reply:
I considered it, but at this time it doesn't really seem worth it. I'll consider adding tests in a future PR especially when things get more interesting (like if we add a bounds, for example)

---
In reply to: [278371092](https://github.com/OfficeDev/office-ui-fabric-react/pull/8426#discussion_r278371092) [](ancestors = 278371092)

Reply:
We're introducing new behavior that folks can easily regress. Are you sure its not worth adding tests to simply verify that the component works?